### ISSUE TITLE
Libhoney should trim events that break limits instead of dropping them

### ISSUE THREAD

**Issue by dcrowe** [honeycombio/libhoney-py#70]
Hi, We recently had a problem where a third-party library was failing with a ludicrously large exception message (>90kb). This is obviously a problem with the third-party library and should be changed, but it also showed us a problem with our honeycomb observability setup.

The problem is that these exceptions weren't turning up. The events were being silently dropped by honeycomb because they were larger than the API limits (https://docs.honeycomb.io/api/events/#body). This lead to us losing some time on incorrect assumptions based on the honeycomb analysis.

I propose that libhoney should attempt to report events that are too large by one of several options:
* trimming them to fit within the limits
* reporting "event too large" events

I think that either option would be better than silence.

In the mean time, I have worked around the issue with the following presend hook:
```
def _presend_hook(event_dict):
    PER_STR_LENGTH_LIMIT = 16 * 1024
    ELLIPSIS = '…'

    for k, v in event_dict.items():
        if isinstance(v, str) and len(v) > PER_STR_LENGTH_LIMIT:
            event_dict[k] = f'{v[:PER_STR_LENGTH_LIMIT - len(ELLIPSIS)]}{ELLIPSIS}'

    return event_dict
```

**Comment by vreynolds** [comment]
This is, unfortunately, not as cut and dry as it may seem. (The "Who's watching the watcher?" problem)

In general, SDKs are quiet by design -- to be respectful of the parent environment. The answer to investigating ingest errors, and I get that it's not ideal, is to watch the [response queue](https://docs.honeycomb.io/getting-data-in/python/beeline/#checking-the-responses-queue). (We should probably make that move obvious in the docs)

In this case, specifically, the API restrictions are intentionally decoupled from the SDK. For instance, the API may change the size requirements, independently of clients.

Out of curiosity: how did you end up discovering the issue with dropped events?

**Comment by dcrowe** [comment]
Agree it's not easy to solve. I've got a work around for us, but I wanted to let you know how this can surprise users.

I discovered the problem when our service stopped delivering data. I went to honeycomb to take a look and start looking at the collected exceptions to see what was going wrong. I quickly found a failure that had only popped up recently and starting working to resolve it. Unfortunately, that lead me in the wrong direction, because the failure that I had found was the wrong one. It was far less urgent than the other one which had not made it to honeycomb. I eventually found the other exception by trawling CloudWatch logs.

After the issue was resolved I then went looking for the reason why it hadn't appeared in honeycomb. Given the absurd size of the exception message, I figured the API limits were good place to start. :)

**Comment by vreynolds** [comment]
Makes sense, thank you for raising this -- definitely an area we'd like to improve.
